### ISSUE TITLE
SA1402 is not raised when declaring multiple types in a file-scoped namespace

### ISSUE THREAD

**Issue by Arthri** [DotNetAnalyzers/StyleCopAnalyzers#3851]
```cs
namespace Z;

public class X {}

public class Y {}
```
doesn't trigger SA1402, but
```cs
public class X {}

public class Y {}
```
and
```cs
namespace Z
{
    public class X {}

    public class Y {}
}
```
do

**Comment by bjornhellander** [comment]
Duplicate of #3435
